parsing/ast_helper.mli

```diff
@@ -303,8 +303,8 @@ module Mod:
     val apply: ?loc:loc -> ?attrs:attrs -> module_expr -> module_expr ->
       module_expr
     val apply_unit: ?loc:loc -> ?attrs:attrs -> module_expr -> module_expr
-    val constraint_: ?loc:loc -> ?attrs:attrs -> module_expr -> module_type ->
-      module_expr
+    val constraint_: ?loc:loc -> ?attrs:attrs -> ?mode:modes -> ?ty:module_type ->
+      module_expr -> module_expr
```

Comment:
It's a bit weird that the "default" way of calling this function&mdash;by omitting all optional arguments&mdash;lets you construct an invalid AST node. I have two alternative ideas:
   * Match what we do with `Exp.constraint_`: just have two required args.
   * Give `constraint_` the type `?loc:loc -> ?attrs:attrs -> ?mode:modes -> module_expr -> module_type -> module_expr`, and introduce a new function `mode_constraint` with optional module type and required modes.